src/audio/modules/passthrough.c

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 NXP. All rights reserved.
+//
+// Author: Paul Olaru <paul.olaru@nxp.com>
+
+#include <stdint.h>
+#include <stddef.h>
+#include <sof/lock.h>
+#include <sof/list.h>
+#include <sof/stream.h>
+#include <sof/alloc.h>
+#include <sof/ipc.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/mixer.h>
+#include <sof/audio/module.h>
+
+#define MODULE_TYPE_PASSTHROUGH 0
+
+struct passthrough_private {
+	void (*fn)(struct comp_dev *dev,
+		   struct comp_buffer *source,
+		   struct comp_buffer *sink,
+		   uint32_t frames);
+	struct comp_buffer *sourceb;
+	struct comp_buffer *sinkb;
+};
+
+static int passthrough_new(struct comp_dev *dev)
+{
+	struct passthrough_private *dat;
+
+	dat = rzalloc(RZONE_RUNTIME, SOF_MEM_CAPS_RAM,
+		      sizeof(struct passthrough_private));
+	if (!dat)
+		return -ENOMEM;
+	module_set_drvdata(dev, dat);
+
+	return 0; /* Success */
+}
+
+static void passthrough_free(struct comp_dev *dev)
+{
+	rfree(module_get_drvdata(dev));
+	module_set_drvdata(dev, NULL);
+}
+
+static int passthrough_params(struct comp_dev *dev)
+{
+	(void)dev;
+	return 0;
+}
+
+static int passthrough_cmd(struct comp_dev *dev, int cmd, void *data,
+			   int max_data_size)
+{
+	(void)dev;
+	(void)cmd;
+	(void)data;
+	(void)max_data_size;
+	return 0;
+}
+
+static void passthrough_copy_16(struct comp_dev *dev,
+				struct comp_buffer *source,
+				struct comp_buffer *sink,
+				uint32_t frames
+			       )
```

Comment:
Checkpatch didn't complain about this lone parenthesis?

Reply:
I may not have some Git hooks configured. Addressed and will look into that.